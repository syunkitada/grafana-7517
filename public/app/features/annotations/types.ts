import { PanelData, DataFrame, AnnotationEvent, TimeRange } from '@grafana/data';
import { DashboardModel, PanelModel } from '../dashboard/state';

export interface AnnotationQueryOptions {
  dashboard: DashboardModel;
  panel: PanelModel;
  range: TimeRange;
}

export interface AnnotationQueryResponse {
  /**
   * All the data flattened to a single frame
   */
  frame?: DataFrame;

  /**
   * The processed annotation events
   */
  events?: AnnotationEvent[];

  /**
   * The original panel response
   */
  panelData?: PanelData;
}
