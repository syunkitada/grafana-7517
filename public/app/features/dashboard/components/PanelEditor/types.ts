import { PanelModel } from '../../state/PanelModel';

export enum DisplayMode {
  Fill = 0,
  Fit = 1,
  Exact = 2,
}

export const displayModes = [
  { value: DisplayMode.Fill, label: 'Fill', description: 'Use all avaliable space' },
  { value: DisplayMode.Fit, label: 'Fit', description: 'Fit in the space keeping ratio' },
  { value: DisplayMode.Exact, label: 'Exact', description: 'Same size as the dashboard' },
];

export enum EditorTab {
  Query = 'query',
  Alerts = 'alerts',
  Transform = 'xform',
}

export const allTabs = [
  { tab: EditorTab.Query, label: 'Query', show: (panel: PanelModel) => true },
  { tab: EditorTab.Alerts, label: 'Alerts', show: (panel: PanelModel) => true },
  { tab: EditorTab.Transform, label: 'Transform', show: (panel: PanelModel) => true },
];
