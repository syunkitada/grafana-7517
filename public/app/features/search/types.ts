export enum DashboardSearchItemType {
  DashDB = 'dash-db',
  DashHome = 'dash-home',
  DashFolder = 'dash-folder',
}

export interface DashboardSection {
  id: number;
  uid?: string;
  title: string;
  expanded?: boolean;
  url: string;
  icon: string;
  score: number;
  hideHeader?: boolean;
  checked?: boolean;
  items: DashboardSectionItem[];
  toggle?: (section: DashboardSection) => Promise<DashboardSection>;
  selected?: boolean;
  type: DashboardSearchItemType;
}

export interface DashboardSectionItem {
  checked?: boolean;
  folderId?: number;
  folderTitle?: string;
  folderUid?: string;
  folderUrl?: string;
  id: number;
  isStarred: boolean;
  selected?: boolean;
  tags: string[];
  title: string;
  type: DashboardSearchItemType;
  uid: string;
  uri: string;
  url: string;
}

export interface DashboardTag {
  term: string;
  count: number;
}

export interface DashboardQuery {
  query: string;
  mode: string;
  tag: string[];
  starred: boolean;
  skipRecent: boolean;
  skipStarred: boolean;
  folderIds: number[];
}

export interface SectionsState {
  sections: DashboardSection[];
  allChecked: boolean;
  dashboardTags: DashboardTag[];
}

export type ItemClickWithEvent = (item: DashboardSectionItem | DashboardSection, event: any) => void;

export type SearchAction = {
  type: string;
  payload?: any;
};

export interface OpenSearchParams {
  query?: string;
}
