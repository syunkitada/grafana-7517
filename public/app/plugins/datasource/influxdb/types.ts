import { DataQuery, DataSourceJsonData } from '@grafana/data';

export interface InfluxOptions extends DataSourceJsonData {
  timeInterval: string;
  httpMode: string;

  // Influx 2.0
  enableFlux?: boolean;
  organization?: string;
  defaultBucket?: string;
  maxSeries?: number;
}

export interface InfluxSecureJsonData {
  password?: string;
  token?: string;
}

export interface InfluxQueryPart {
  type: string;
  params?: string[];
  interval?: string;
}

export interface InfluxQueryTag {
  key: string;
  operator?: string;
  condition?: string;
  value: string;
}

export enum InfluxQueryType {
  Classic = 'Classic', // IFQL query builder
  InfluxQL = 'InfluxQL', // raw ifql
  Flux = 'Flux',
}

export interface InfluxQuery extends DataQuery {
  queryType?: InfluxQueryType;
  policy?: string;
  measurement?: string;
  resultFormat?: 'time_series' | 'table';
  orderByTime?: string;
  tags?: InfluxQueryTag[];
  groupBy?: InfluxQueryPart[];
  select?: InfluxQueryPart[][];
  limit?: string;
  slimit?: string;
  tz?: string;
  fill?: string;
  rawQuery?: boolean; // deprecated (use raw InfluxQL)
  query?: string;
}
