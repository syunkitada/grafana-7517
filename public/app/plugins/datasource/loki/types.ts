import { DataQuery } from '@grafana/ui/src/types';

export interface LokiQuery extends DataQuery {
  expr: string;
}

