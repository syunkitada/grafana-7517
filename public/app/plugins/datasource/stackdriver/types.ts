export enum MetricFindQueryTypes {
  Services = 'services',
  MetricTypes = 'metricTypes',
  MetricLabels = 'metricLabels',
  ResourceLabels = 'resourceLabels',
  ResourceTypes = 'resourceTypes',
  Aggregations = 'aggregations',
  Alignerns = 'alignerns',
  AlignmentPeriods = 'alignmentPeriods',
}

export interface TemplateQueryComponentData {
  selectedQueryType: string;
  metricDescriptors: any[];
  selectedService: string;
  selectedMetricType: string;
  labels: string[];
  labelKey: string;
  metricTypes: Array<{ value: string; name: string }>;
  services: Array<{ value: string; name: string }>;
}
