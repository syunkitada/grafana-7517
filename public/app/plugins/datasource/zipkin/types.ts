export type ZipkinSpan = {
  traceId: string;
  parentId?: string;
  name: string;
  id: string;
  timestamp: number;
  duration: number;
  localEndpoint: {
    serviceName: string;
    ipv4: string;
    port?: number;
  };
  annotations?: ZipkinAnnotation[];
  tags?: { [key: string]: string };
  kind?: 'CLIENT' | 'SERVER' | 'PRODUCER' | 'CONSUMER';
};

export type ZipkinAnnotation = {
  timestamp: number;
  value: string;
};
