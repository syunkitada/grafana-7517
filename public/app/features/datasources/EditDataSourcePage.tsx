import React, { PureComponent } from 'react';
import { hot } from 'react-hot-loader';
import { connect } from 'react-redux';
import PageHeader from '../../core/components/PageHeader/PageHeader';
import { DataSource, NavModel } from 'app/types';
import { loadDataSource } from './state/actions';
import { getNavModel } from '../../core/selectors/navModel';
import { getRouteParamsId, getRouteParamsPage } from '../../core/selectors/location';
import { getDataSourceLoadingNav } from './state/navModel';
import { getDataSource } from './state/selectors';

export interface Props {
  navModel: NavModel;
  dataSource: DataSource;
  dataSourceId: number;
  pageName: string;
  loadDataSource: typeof loadDataSource;
}

enum PageTypes {
  Settings = 'settings',
  Permissions = 'permissions',
  Dashboards = 'dashboards',
}

export class EditDataSourcePage extends PureComponent<Props> {
  componentDidMount() {
    this.fetchDataSource();
  }

  async fetchDataSource() {
    await this.props.loadDataSource(this.props.dataSourceId);
  }

  isValidPage(currentPage) {
    return (Object as any).values(PageTypes).includes(currentPage);
  }

  getCurrentPage() {
    const currentPage = this.props.pageName;

    return this.isValidPage(currentPage) ? currentPage : PageTypes.Permissions;
  }

  renderPage() {
    switch (this.getCurrentPage()) {
      case PageTypes.Permissions:
        return <div>Permissions</div>;
    }

    return null;
  }

  render() {
    const { navModel } = this.props;

    return (
      <div>
        <PageHeader model={navModel} />
        <div className="page-container page-body">{this.renderPage()}</div>
      </div>
    );
  }
}

function mapStateToProps(state) {
  const pageName = getRouteParamsPage(state.location) || PageTypes.Permissions;
  const dataSourceId = getRouteParamsId(state.location);
  const dataSourceLoadingNav = getDataSourceLoadingNav(pageName);

  return {
    navModel: getNavModel(state.navIndex, `datasource-${pageName}-${dataSourceId}`, dataSourceLoadingNav),
    dataSourceId: dataSourceId,
    dataSource: getDataSource(state.dataSources, dataSourceId),
    pageName: pageName,
  };
}

const mapDispatchToProps = {
  loadDataSource,
};

export default hot(module)(connect(mapStateToProps, mapDispatchToProps)(EditDataSourcePage));
