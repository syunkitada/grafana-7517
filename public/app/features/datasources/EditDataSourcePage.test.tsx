import React from 'react';
import { shallow } from 'enzyme';
import { EditDataSourcePage, Props } from './EditDataSourcePage';
import { DataSource, NavModel } from '../../types';

const setup = (propOverrides?: object) => {
  const props: Props = {
    navModel: {} as NavModel,
    dataSource: {} as DataSource,
    dataSourceId: 1,
    pageName: '',
    loadDataSource: jest.fn(),
  };

  Object.assign(props, propOverrides);

  const wrapper = shallow(<EditDataSourcePage {...props} />);
  const instance = wrapper.instance() as EditDataSourcePage;

  return {
    wrapper,
    instance,
  };
};

describe('Render', () => {
  it('should render component', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should render permissions page', () => {
    const { wrapper } = setup({
      pageName: 'permissions',
    });

    expect(wrapper).toMatchSnapshot();
  });
});

describe('Functions', () => {
  describe('is page valid', () => {
    it('should be a valid page', () => {
      const { instance } = setup();

      expect(instance.isValidPage('permissions')).toBeTruthy();
    });

    it('should not be a valid page', () => {
      const { instance } = setup();

      expect(instance.isValidPage('asdf')).toBeFalsy();
    });
  });

  describe('get current page', () => {
    it('should return permissions', () => {
      const { instance } = setup({
        pageName: 'permissions',
      });

      expect(instance.getCurrentPage()).toEqual('permissions');
    });

    it('should return settings if bogus route', () => {
      const { instance } = setup({
        pageName: 'asdf',
      });

      expect(instance.getCurrentPage()).toEqual('settings');
    });
  });
});
