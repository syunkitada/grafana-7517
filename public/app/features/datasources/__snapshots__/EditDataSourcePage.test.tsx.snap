// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
<div>
  <PageHeader
    model={Object {}}
  />
  <div
    className="page-container page-body"
  >
    <Connect(DataSourceSettings) />
  </div>
</div>
`;

exports[`Render should render permissions page 1`] = `
<div>
  <PageHeader
    model={Object {}}
  />
  <div
    className="page-container page-body"
  >
    <Connect(DataSourcePermissions) />
  </div>
</div>
`;
